src/gui/widgets/Knob.cpp

```diff
@@ -839,3 +832,37 @@ void Knob::doConnections()
 						this, SLOT( update() ) );
 	}
 }
+
+
+void Knob::changeEvent(QEvent * ev)
+{
+	if (ev->type() == QEvent::EnabledChange)
+	{
+		onKnobNumUpdated();
+		if (!m_label.isEmpty())
+		{
+			setLabel(m_label);
+		}
+		m_cache = QImage();
+		update();
+	}
+}
+
+
+void convertPixmapToGrayScale(std::unique_ptr<QPixmap> &pixMap)
+{
+	QImage temp = pixMap->toImage().convertToFormat(QImage::Format_ARGB32);
+	for (int i = 0; i < temp.height(); ++i)
+	{
+		for (int j = 0; j < temp.width(); ++j)
+		{
+			QColor pix;
+			pix = temp.pixelColor(i, j);
+			quint8 gscale = quint8((0.2126*pix.red() + 0.7152*pix.green() + 0.0722*pix.blue()));
+			QRgba64 pix_gray64;
+			pix_gray64 = QRgba64::fromRgba(gscale, gscale, gscale, quint8(pix.alpha()));
+			temp.setPixelColor(i, j, pix_gray64);
```

Comment:
These variables can be declared and initialised in the same statement. Also, `QColor` supports the use of floating-point colour components, which would remove the need for casts.
```suggestion
			const auto pix = temp.pixelColor(i, j);
			const auto gscale = 0.2126 * pix.redF() + 0.7152 * pix.greenF() + 0.0722 * pix.blueF();
			const auto pixGray = QColor::fromRgbF(gscale, gscale, gscale, pix.alphaF());
			temp.setPixelColor(i, j, pixGray);
```